sklearn/tests/test_calibration.py

```diff
@@ -275,6 +275,34 @@ def multiclass_brier(y_true, proba_pred, n_classes):
     assert calibrated_brier < 1.1 * uncalibrated_brier
 
 
+def test_calibration_zero_probability():
+    class FakeClassifier():
+        def __init__(self, n_classes):
+            self.classes_ = n_classes
+
+        def predict_proba(self, X):
+            return np.empty_like(X)
+
+    class FakeCalibrator():
+        # This function is called from _CalibratedClassifier.predict_proba.
+        def predict(self, X):
+            return np.zeros_like(X)
+
+    X, y = make_blobs(n_samples=500, n_features=100, random_state=7,
+                      centers=10, cluster_std=15.0)
+    n_classes = np.unique(y)
```

Comment:
In scikit-learn classifiers, this operation is typically performed in the fit method of `FakeClassifier` to define the `self.classes_` attribute with the list of unique classes. Then you would use it in predict proba to return `np.empty(shape=(X.shape[0], self.classes_.shape[0]))`.

And then you do would no longer need to pass `n_classes` to `__init__` and therefore do not need to define `FakeClassifier.__init__`

Reply:
This would be even simpler to just use a `DummyClassifier`:

https://github.com/scikit-learn/scikit-learn/pull/19421#discussion_r573753465

assuming they implement either `predict_proba` or `decision_function`.

Reply:
Thanks, I used `DummyCallsifier` and removed `FakeClassifier`.

Reply:
This line can still be removed because the list of unique classes should already be computed when fitting the `DummyClassifier`: https://github.com/scikit-learn/scikit-learn/pull/19421/files#r573825143

Reply:
I removed this line completely, and use `cls.classes_` and `cls.n_classes_` instead.